### PR TITLE
Operational policy improvements to align with APIM 4.1.0 beta

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
- Process JSON policy spec
- Refactor rename policy In, Out as Request, Response
- Remove regex and type validation of operation policies using spec
- Handle includes as comma separated string in call_interceptor policy
- Handle additional attributes of OPA policy
    - handle, "additionalProperties", "maxOpenConnections", "maxPerRoute", "connectionTimeout"
- Update OPA request payload to the server
- Handle default value in policy spec

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2710

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/policy_container.go [hunk 1]
```diff
@@ -182,70 +182,38 @@ func (spec *PolicySpecification) validatePolicy(policy Policy, flow PolicyFlow,
...
 			}
-			loggers.LoggerOasparser.Warnf("Operation policy \"%v\" not allowed in multiple times, appling the first policy", policy.PolicyName)
+			loggers.LoggerOasparser.Warnf("Operation policy \"%s\" not allowed in multiple times, appling the first policy", policy.PolicyName)
 		}
 	}
 
 	policyPrams, ok := policy.Parameters.(map[string]interface{})
 	if ok {
 		for _, attrib := range spec.Data.PolicyAttributes {
-			val, found := policyPrams[attrib.Name]
```
Comment (AmaliMatharaarachchi) [inline adapter/internal/oasparser/model/policy_container.go:200]: I see that we have removed attrib validation for policies. Don't we need to?

Comment (renuka-fernando) [reply adapter/internal/oasparser/model/policy_container.go:200]: it is in the line 200 of newly added changes

Comment (suksw) [reply adapter/internal/oasparser/model/policy_container.go:200]: Will that be enough? Are you sure about removing what you already had used?

File: enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/opa/OPAConstants.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.choreo.connect.enforcer.commons.opa;
+
+/**
+ * This class represents the constants that are used for OPA mediator
```
Comment (praminda) [inline enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/opa/OPAConstants.java:21]:
```suggestion
 * This class represents the constants that are used for OPA mediator.
```

File: enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/opa/OPAConstants.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+public class OPAConstants {
+    /**
+     * Constants of the AdditionalParameters map
```
Comment (praminda) [inline enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/opa/OPAConstants.java:25]:
```suggestion
     * Constants of the AdditionalParameters map.
```

File: adapter/internal/oasparser/model/policy_container.go (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply adapter/internal/oasparser/model/policy_container.go:200]: seems like we have removed attrib key validation, that's what I meant

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**